sdk/trace/trace_test.go

```diff
@@ -93,18 +93,21 @@ func TestSetName(t *testing.T) {
 			sampledBefore: true,
 			sampledAfter:  false,
 		},
-		{ // 2
-			name:          "barbar",
-			newName:       "barbaz",
-			sampledBefore: false,
-			sampledAfter:  false,
-		},
-		{ // 3
-			name:          "barbar",
-			newName:       "foobar",
-			sampledBefore: false,
-			sampledAfter:  true,
-		},
+		// TODO (paivagustavo): this test used `WithRecord` to force the span to have
+		//  a export data, this ensured that `SetName` worked nicely, but `WithRecord`
+		//  was removed from the API.
+		//{ // 2
```

Comment:
Lets' open another issue to track this. One possible solution is listed [here](https://github.com/open-telemetry/opentelemetry-go/pull/223#issuecomment-543843322) but it requires Sampling parameter to have some indication that the sampler is being called again.

Reply:
Ok, I'll create another issue to track this. I think we need to discuss a little further about the problem, I'll try to explain the consequences of doing this.